src/main/java/org/example/service/shoppingcart/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,99 @@
+package org.example.service.shoppingcart;
+
+import lombok.RequiredArgsConstructor;
+import org.example.dto.cartItem.CartItemRequestDto;
+import org.example.dto.cartItem.CartItemUpdateDto;
+import org.example.dto.shoppingcart.ShoppingCartDto;
+import org.example.exception.EntityNotFoundException;
+import org.example.mapper.CartItemMapper;
+import org.example.model.Book;
+import org.example.model.CartItem;
+import org.example.model.User;
+import org.example.model.ShoppingCart;
+import org.example.repository.BookRepository;
+import org.example.repository.CartItemRepository;
+import org.example.repository.ShoppingCartRepository;
+import org.example.repository.UserRepository;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import java.util.List;
+
+@Service
+@Transactional
+@RequiredArgsConstructor
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final UserRepository userRepository;
+    private final BookRepository bookRepository;
+    private final CartItemRepository cartItemRepository;
+    private final CartItemMapper cartItemMapper;
+
+    @Override
+    public ShoppingCartDto getByUserId(Long authenticationId) {
+        ShoppingCart shoppingCart = shoppingCartRepository.findByUserId(authenticationId)
+                .orElseThrow(() -> new EntityNotFoundException("Can't get user by Id " + authenticationId));
+
+        return cartItemMapper.toDto(shoppingCart);
```

Comment:
```suggestion
        return shoppingCartRepository.findByUserId(authenticationId)
                .map(cartItemMapper::toDto)
                .orElseThrow(() -> new EntityNotFoundException("Can't get user by Id " + authenticationId));
```
fix same issue in all places